e2e/libs/node/operations/aws.py

```diff
@@ -0,0 +1,103 @@
+from node.operations.abstract_cloud_provider import AbstractCloudProvider
+from node import Nodes
+
+import boto3
+import logging
+
+class EC2(AbstractCloudProvider):
+
+    _ec2_instance = boto3.resource('ec2')
+
+    def __init__(self) -> None:
+        super().__init__()
+
+    def get_all_node_instances(self):
+        result = []
+        for node in Nodes.all_nodes:
+            instances = self.get_node_instance(node['name'])
+            result.append(instances)
+        return result
+
+    def get_node_instance(self, node_name):
+        if node_name == "":
+            raise Exception(f"empty parameter: node_name={node_name}")
```

Comment:
```suggestion
            raise Exception("empty parameter: node_name")
```

or 

```suggestion
            raise Exception(f"invalid parameter: node_name={node_name}")
```

And suggest to extract to a reusable function.

Reply:
Actually, do we still need to check this? IIRC its been checked on the caller already?